python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -476,8 +477,39 @@ def runQuantum(
             task.runQuantum(butlerQC, inputRefs, outputRefs)
         except NoWorkFound as err:
             # Not an error, just an early exit.
-            _LOG.info("Task '%s' on quantum %s exited early: %s", task_node.label, quantum.dataId, str(err))
-            pass
+            _LOG.info(
+                "Task '%s' on quantum %s exited early with no work found: %s.",
+                task_node.label,
+                quantum.dataId,
+                str(err),
+            )
+        except AnnotatedPartialOutputsError as caught:
+            error: BaseException
+            if caught.__cause__ is None:
+                _LOG.warning(
+                    "Incorrect use of AnnotatedPartialOutputsError: no chained exception found.",
+                    task_node.label,
+                    quantum.dataId,
+                )
```

Comment:
Oh, I like that. I think it should maybe be an `error`, though, not a `warning`, so it's more visible? That seems like a serious coding problem that should be loud.

Reply:
I don't really like ERROR or higher log messages that don't correspond to an exception or some other actual execution stoppage.  I thought about making this a hard failure, but I figured we'd be much more frustrated if it brought down processing that was otherwise working as intended.

Reply:
There are ERROR, FAILURE, EXCEPTION log types; I wish we'd used FAILURE for what you're describing, so that we can use ERROR for other things ("be louder than warning, but I (maybe) didn't stop running").

Reply:
Note that `exception` is not a distinct log level; it's just [`error` with automatic stack trace printing](https://docs.python.org/3/library/logging.html#logging.Logger.exception).